### PR TITLE
Use hw card reader by default

### PR DESCRIPTION
**Pull request by malinajirka**
Parent issue #3725 

This PR changes the default card reader from simulated to a hw reader. Since we have implemented the vital flows, using a real hw reader might be preferred so we can notice some edge case issues. 

To test:
Try to connect to a hw reader.

@nbradbury @kidinov I'm explicitly requesting your review to make sure you both know about this change in the default behavior.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by kidinov** [comment]
Maybe to add a flag that we can pass via command line params? We can keep it by default `false`, but when needed to quickly test something with simulated reader we will have an opportunity to pass it.

For instance:
```
./gradlew install .. -PuseSimulatedReader 
```

Then in buildConfigField to reference those 

```
buildConfigField "boolean", "useSimulatedReader", System.getProperty("useSimulatedReader"))
```

**Comment by malinajirka** [comment]
Great idea! @kidinov  ❤️ !!!

I've introduced a gradle project property in gradle.properties. It can be overridden using `-PuseSimulatedReader=true` command line flag.

To test:
1. Build the project and notice BuildConfig.USE_SIMULATED_READER is `false`
2. Build project  with the command line flag `./gradlew buildWasabiDebug -PuseSimulatedReader=true` and notice the field is `true` now


Note: When this change gets reviewed, we'll need to update CI gradle.properties as otherwise the build will fail, since the property will not be defined. I didn't want to do this until the changes are reviewed. We'll also need to give a headsup to other android devs to copy the new property from the example file.

cc @jkmassel since we are touching gradle.properties and build.gradle.

File: WooCommerce/build.gradle [hunk 1]
```diff
@@ -66,6 +66,8 @@ android {
         testInstrumentationRunner 'com.woocommerce.android.WooCommerceTestRunner'
         // TODO remove this once the hilt migration is complete
         javaCompileOptions.annotationProcessorOptions.arguments['dagger.hilt.disableModulesHaveInstallInCheck'] = 'true'
+
+        buildConfigField "boolean", "USE_SIMULATED_READER", useSimulatedReader
```
Comment (kidinov) [inline WooCommerce/build.gradle]: I would go with just (without modification of `gradle.properties`)

```
buildConfigField "boolean", "USE_SIMULATED_READER", project.hasProperty("useSimulatedReader").toString()
```

* We don't break the build for others
* No need to modify the CI
* We still can use simulator via build flag (also possible to set a flag in AS)
* Also a bit easier to delete this when we won't need that

Comment (kidinov) [reply WooCommerce/build.gradle]: Then we can just run

```
./gradlew clean installJalapenoDebug -PuseSimulatedReader
```

or add a flag in AS

image

Comment (malinajirka) [reply WooCommerce/build.gradle]: I usually like to keep user-properties in gradle.properties file for "documentation purposes". However, the benefits you outlined make total sense. Moreover, we can always add it to gradle.properties if this flag becomes a stable flag. Good call, thank you for the suggestion!

**Review by kidinov — APPROVED** [review]
